src/config.rs

```diff
@@ -92,30 +95,28 @@ impl Config {
     pub fn save_to_path(&self, path: &Path) {
         Self::create_dir(path);
 
-        let file = OpenOptions::new()
-            .write(true)
-            .create(true)
-            .truncate(true)
-            .open(path)
-            .unwrap();
-
-        let writer = io::BufWriter::new(file);
-        serde_json::to_writer(writer, self).unwrap();
+        fs::write(path, toml::to_string(self).unwrap()).unwrap();
     }
 
     pub fn set_watch(&mut self, path: String, cfg: WatchConfig) {
-        if self.repos.contains_key(&path) {
-            println!("{} is already being watched", path)
+        let abs_path = fs::canonicalize(path).expect("The provided path is not a directory");
+        let abs_path = abs_path.to_str().unwrap();
+
+        if self.repos.contains_key(abs_path) {
+            println!("{} is already being watched", abs_path)
         } else {
-            self.repos.insert(path.clone(), Rc::new(cfg));
-            println!("Started watching {}", path)
+            self.repos.insert(abs_path.to_string(), Rc::new(cfg));
```

Comment:
This still does a clone, yeah? How else could it produced a new owned object?

Reply:
To my knowledge, `to_string()` will move and effectively consume the value, but doesn't clone it. Since the `watch` and `unwatch` funcs are fire-and-forget, and it's never any more than a single string, I'd regard it as less of a concern anyway.